### DISCUSSION TITLE
流程控制

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#164] [Q&A] [answered]
在使用if...elif...else練習流程控制時，我碰到下面的情況。我猜想問題，就是在討論區(Discussions)曾經討論過，所謂電腦儲存「浮點數(float)」會有誤差的問題。但，我目前不知道怎麼解決。若目前的程式，使用在網路購物上，就會出現問題。
![image](https://user-images.githubusercontent.com/86505454/142757752-f1ba6fc3-dfdc-4e49-9a17-0f830e28c0d5.png)

**Comment by EricAgumonJou** [comment]
這個確實是浮點數誤差的關係（我的電腦也是這樣顯示）
<img width="1440" alt="Screen Shot 2021-11-24 at 15 04 32" src="https://user-images.githubusercontent.com/91121500/143191005-d4caeb65-2d44-4007-b2ae-2ed2208f637d.png">

**Comment by EricAgumonJou** [comment, marked as answer]
若要處理這個數據，例如取到小數點第2位，我print的地方會打：
print('{:.2f}'.format(discount))

**Reply by enoch-lightning** [reply]
文字格式化應該是一種解決方式，因為整數位、十分位、百分位、千分位、萬分位、十萬分位、百萬分位、千萬分位、億分位、十億分位，或甚至更大的分位都可以用`:.想留幾位小數f`來保留，並限制小數位數。
